DegreeProject/Assets/Scripts/Enemy/States/PatrolState.cs

```diff
@@ -76,24 +86,52 @@ private void HorizontalCollision(AI owner)
         {
             Vector2 direction = Vector2.right;
 
-            RaycastHit2D hit = Physics2D.BoxCast(originRight, boxSize, 0, direction, distance, LayerMask.GetMask(StringData.groundLayer));
+            RaycastHit2D hit = Physics2D.BoxCast(originRight, boxSize, 0f, direction, distance, LayerMask.GetMask(ground));
+
+            // On the right side of the enemy the distance of the ray needs to be longer because otherwise it wont hit
+            RaycastHit2D hitE = Physics2D.Raycast(originRight, direction, 0.3f, LayerMask.GetMask(enemy));
 
             if (hit)
             {
                 rightWall = true;
-            } else { rightWall = false; }
+            }
+            else { rightWall = false; }
+
+            var otherEnemy = hitE.collider;
+
+            if (otherEnemy != null)
+            {
+                if (otherEnemy.name != owner.name)
+                {
+                    rightWall = true;
+                }
+                else { rightWall = false; }
+            }
         }
 
         // Left 
```

Comment:
Här gör du samma grej som för // Right
Du skulle kunna baka ihop det här till en metod istället, som returnerar en bool, och tar in direction m.m.
Ju mindre kod du har, desto färre chanser för buggar.